### DISCUSSION TITLE
Assets: how to retrieve Name, Unit and URL?

### DISCUSSION THREAD

**Discussion by FerGT50** [CareBoo/unity-algorand-sdk#196] [Q&A] [answered]
Hello,
using your 4.0.0-pre branch with Unity 2021.3.19 LTS; thanks for your excellent work!

I need to retrieve assets properties, so after retrieving the account from the indexer and checking the balance, I scan the retrieved AssetHolding array.
Problem is, the AssetHolding struct does not seem to contain all asset properties: Asset Name, Asset Unit and Asset URL are missing (while AssetID and Asset Amount are available):
https://careboo.github.io/unity-algorand-sdk/4.0/api/Algorand.Unity.Indexer.AssetHolding.html

How should I retrieve Name, Unit and URL?

Thanks,
Fernando

**Comment by jasonboukheir** [comment, marked as answer]
Hi Fernando,

You can obtain the asset details with the following indexer API:
[LookupAssetByID](https://careboo.github.io/unity-algorand-sdk/4.0/api/Algorand.Unity.IndexerClient.html#Algorand_Unity_IndexerClient_LookupAssetByID_System_UInt64_Algorand_Unity_Optional_System_Boolean__)

Usually, obtaining asset details for a user first involves obtaining their `AssetHolding`s, then using the ids there to obtain details on those assets. If you're looking to get an array of assets, you can alternatively use: [SearchForAssets](https://careboo.github.io/unity-algorand-sdk/4.0/api/Algorand.Unity.IndexerClient.html#Algorand_Unity_IndexerClient_SearchForAssets_Algorand_Unity_Optional_System_Boolean__Algorand_Unity_Optional_System_UInt64__System_String_System_String_System_String_System_String_Algorand_Unity_Optional_System_UInt64__)

**Reply by FerGT50** [reply]
Thank you Jason,

by any chance, do you have an example showing a call to LookupAssetByID? To better understand how to parse the result and extract say the asset name.

Thank you very much

**Reply by jasonboukheir** [reply]
> by any chance, do you have an example showing a call to LookupAssetByID? To better understand how to parse the result and extract say the asset name.

Not at the moment, but I'll be sure to keep this in mind for some samples. This seems to be one of the most common parts that people struggle with.

To keep it brief, here's a quick example:

```csharp
AssetIndex myAssetId = 1001;
IndexerClient indexer = new IndexerClient(/*...*/);
var (lookupAssetResponse, lookupAssetError) = await indexer.LookupAssetByID(myAssetId);
if (lookupAssetError)
{
  Debug.LogError(lookupAssetError);
}
else
{
  var asset = lookupAssetResponse.Asset;
  Debug.Log($"my asset name is {asset.Params.Name}");
}
```

**Reply by FerGT50** [reply]
Thank you for following up (and congratulations for 4.0.0 release!).

Unfortunately I cannot get your example to work (using 4.0.0 release, Unity 2021.3.19 LTS). 

Looks like lookupAssetError is not a bool (and does not even accept a cast to bool) and lookupAssetResponse has no "Asset" property:

`Error	CS0029	Cannot implicitly convert type 'Algorand.Unity.Indexer.AssetResponse' to 'bool'`

and

`Error	CS1061	'ErrorResponse' does not contain a definition for 'Asset' and no accessible extension method 'Asset' accepting a first argument of type 'ErrorResponse' could be found (are you missing a using directive or an assembly reference?)`

**Reply by FerGT50** [reply]
Mmm, looks like it should be

`var (lookupAssetError, lookupAssetResponse) = await indexer.LookupAssetByID(myAssetId);`

( not var (lookupAssetResponse, lookupAssetError) = await indexer.LookupAssetByID(myAssetId); )
